Stubgen import tests fail when run sequentially

This causes 11 test failures for me:

```
$ pytest mypy/test/teststubgen.py -n0 -k _import
...
__________________________________________ testCommentForUndefinedName_import __________________________________________
data: /Users/jukka/src/mypy/test-data/unit/stubgen.test:504:
/Users/jukka/src/mypy/mypy/test/teststubgen.py:279: in run_case
    self.add_file(os.path.join(out_dir, 'main.pyi'), a)
/Users/jukka/src/mypy/mypy/test/teststubgen.py:300: in add_file
    with open(path, encoding='utf8') as file:
E   FileNotFoundError: [Errno 2] No such file or directory: 'out/main.pyi'
------------------------------------------------- Captured stdout call -------------------------------------------------
Failed to import main; skipping it
====================================== 11 failed, 162 deselected in 1.01 seconds =======================================
```

However, if I run tests in parallel, they pass:

```
$ pytest mypy/test/teststubgen.py -n2 -k _import
...
============================================== 11 passed in 1.46 seconds ===============================================
```

--- a/mypy/test/teststubgen.py
+++ b/mypy/test/teststubgen.py
@@ -256,6 +256,15 @@ class StubgenPythonSuite(DataSuite):
     files = ['stubgen.test']
 
     def run_case(self, testcase: DataDrivenTestCase) -> None:
+        old_sys_path = sys.path[:]
+        if not ('' in sys.path or '.' in sys.path):
+            sys.path.insert(0, '')
+        try:
+            self.run_case_inner(testcase)
+        finally:
+            sys.path = old_sys_path
+
+    def run_case_inner(self, testcase: DataDrivenTestCase) -> None:
         extra = []
         mods = []
         source = '\n'.join(testcase.input)

